packages/babel-parser/src/parser/lval.js

```diff
@@ -60,6 +60,14 @@ export default class LValParser extends NodeUtils {
             const prop = node.properties[i];
             const isLast = i === last;
             this.toAssignableObjectExpressionProp(prop, isBinding, isLast);
+
+            if (
+              isLast &&
+              prop.type === "RestElement" &&
+              node.extra?.trailingComma
+            ) {
+              this.raiseRestNotLast(node.extra.trailingComma);
```

Comment:
By raising error at comma position we are not really consistent with where this error is throw in other situation, i.e. in `toAssignableObjectExpressionProp`.

https://github.com/babel/babel/blob/11fa2461cec930168fd43cfeda733aab056d8318/packages/babel-parser/src/parser/lval.js#L132-L134

Hence we could throw at `prop.start`.